motion_predict/CMakeLists.txt

```diff
@@ -13,89 +13,53 @@
 # License for the specific language governing permissions and limitations under
 # the License.
 
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(motion_predict)
 
 find_package(carma_cmake_common REQUIRED)
-carma_check_ros_version(1)
+carma_check_ros_version(2)
 carma_package()
 
-## Find catkin macros and libraries
-## if COMPONENTS list like find_package(catkin REQUIRED COMPONENTS xyz)
-## is used, also find other catkin packages
-find_package(catkin REQUIRED COMPONENTS
-  cav_msgs
-  roscpp
-)
-
-## System dependencies are found with CMake's conventions
+find_package(ament_cmake_auto REQUIRED)
 find_package(Eigen3 REQUIRED)
-
-
-###################################
-## catkin specific configuration ##
-###################################
-## The catkin_package macro generates cmake config files for your package
-## Declare things to be passed to dependent projects
-## INCLUDE_DIRS: uncomment this if your package contains header files
-## LIBRARIES: libraries you create in this project that dependent projects also need
-## CATKIN_DEPENDS: catkin_packages dependent projects also need
-## DEPENDS: system dependencies of this project that dependent projects also need
-catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES motion_predict
-  CATKIN_DEPENDS cav_msgs roscpp
-  DEPENDS EIGEN3 # Eigen is a special case where is needs to be in caps here but not in find_package
+ament_auto_find_build_dependencies()
+
+set(dependencies
+    rclcpp
+    rclcpp_components
+    carma_perception_msgs
+    geometry_msgs
+    EIGEN3
 )
 
-###########
-## Build ##
-###########
-
-## Specify additional locations of header files
-## Your package locations should be listed before other locations
+# Includes
 include_directories(
   include
-  ${catkin_INCLUDE_DIRS}
-  ${EIGEN3_INCLUDE_DIRS}
 )
 
-add_library(motion_predict 
+set(cpp_files
   src/motion_predict.cpp
   src/predict_ctrv.cpp
 )
 
-add_dependencies(motion_predict ${catkin_EXPORTED_TARGETS})
+carma_define_ament_library(${PROJECT_NAME} "${dependencies}" "${cpp_files}")
```

Comment:
Why not just us ament_auto_add_library() instead of these custom macros here? Then you wouldn't need to list out the dependencies in the cmake